fastapi/middleware/asyncexitstack.py

```diff
@@ -0,0 +1,28 @@
+from typing import Optional
+
+from fastapi.concurrency import AsyncExitStack
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+
+class AsyncExitStackMiddleware:
+    def __init__(self, app: ASGIApp, context_name: str = "fastapi_astack") -> None:
+        self.app = app
+        self.context_name = context_name
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+        if AsyncExitStack:
```

Comment:
It seems to me that `AsyncExitStack` is always true, it’s imported and concurrency provides it from either of two locations, but it’s always provided.

Reply:
yeah... :thinking: